docs/v0.6/en/user-guide/client-libraries/go.md

```diff
@@ -36,129 +37,232 @@ import (
+err = cpuMetric.AddRow("127.0.0.1", time.Now(), 0.1, 0.12)
+if err != nil {
+    // Handle error appropriately
+}
 
-client, _ := greptime.NewClient(cfg)
 ```
 
 %}
 
+{template create-rows%
 
+```go
+cpuMetric, err := table.New("cpu_metric")
+if err != nil {
+    // Handle error appropriately
+}
+cpuMetric.AddTagColumn("host", types.STRING)
+cpuMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+cpuMetric.AddFieldColumn("cpu_user", types.FLOAT)
+cpuMetric.AddFieldColumn("cpu_sys", types.FLOAT)
+err = cpuMetric.AddRow("127.0.0.1", time.Now(), 0.1, 0.12)
+if err != nil {
+    // Handle error appropriately
+}
 
-{template row-object%
+memMetric, err := table.New("mem_metric")
+if err != nil {
+    // Handle error appropriately
+}
+memMetric.AddTagColumn("host", types.STRING)
+memMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+memMetric.AddFieldColumn("mem_usage", types.FLOAT)
+err = memMetric.AddRow("127.0.0.1", time.Now(), 112)
+if err != nil {
+    // Handle error appropriately
+}
+```
+
+%}
 
-The Go ingester SDK uses `Series` to represent a row data item. Multiple `Series` can be added to a `Metric` object and then written to GreptimeDB.
+{template insert-rows%
+
+```go
+resp, err := cli.Write(context.Background(), cpuMetric, memMetric)
+if err != nil {
+    // Handle error appropriately
+}
+log.Printf("affected rows: %d\n", resp.GetAffectedRows().GetValue())
+```
 
 %}
 
+{template streaming-insert%
+
+To insert data using the streaming API, you must first create a stream client.
+
+```go
+cfg := config.New("<host>").
+    WithAuth("<username>", "<password>").
+    WithDatabase(database)
+streamClient, err := client.NewStreamClient(cfg)
+```
 
-{template create-a-rows%
+Then send the data to the server.
 
 ```go
-seriesHost1 := greptime.Series{}
-seriesHost1.AddStringTag("host", "host1")
-seriesHost1.AddFloatField("cpu", 0.90)
-seriesHost1.SetTimestamp(time.Now())
+err := streamClient.Send(context.Background(), cpuMetric, memMetric)
+if err != nil {
+    // Handle error appropriately
+}
 ```
 
 %}
 
-{template create-rows%
+{template update-rows%
 
 ```go
-seriesHost1 := greptime.Series{}
-seriesHost1.AddStringTag("host", "host1")
-seriesHost1.AddFloatField("cpu", 0.90)
-seriesHost1.SetTimestamp(time.Now())
-
-seriesHost2 := greptime.Series{}
-seriesHost2.AddStringTag("host", "host2")
-seriesHost2.AddFloatField("cpu", 0.70)
-seriesHost2.SetTimestamp(time.Now())
-
-metric := greptime.Metric{}
-metric.AddSeries(seriesHost1, seriesHost2)
+ts := time.Now()
+_ = cpuMetric.AddRow("127.0.0.1", ts, 0.1, 0.12)
+// insert a row data
+resp, _ := cli.Write(context.Background(), cpuMetric)
+
+// update the row data
+// The same tag `127.0.0.1`
+// The same time index `ts`
+// The new value: cpu_user = `0.80`, cpu_sys = `0.11`
+_ = cpuMetric.AddRow("127.0.0.1", ts, 0.80, 0.11)
+// overwrite the existing data
+resp, _ = cli.Write(context.Background(), cpuMetric)
 ```
 
 %}
 
 
-{template save-rows%
+{template orm-style-object%
```

Comment:
If `orm-style-object` is not recommended in docs, maybe we can make a refer link to ingester-go, make it easier to let user explore the reference of the original documentation in GitHub or somewhere else.

For too many duplicated codes in docs  may confuse users.

Reply:
As SDKer discussed, we will reserve low-level and high-level(ORM) code in the user documentation.